test/modules/TestResultCoordinator/TestReportUtil.cs

```diff
@@ -103,9 +110,100 @@ internal static List<ITestReportMetadata> ParseReportMetadataJson(string reportM
             return reportMetadataList;
         }
 
+        internal static async Task<Uri> GetOrCreateBlobContainerSasUriForLogAsync(string storageAccountConnectionString)
+        {
+            string containerName = GetAzureBlobContainerNameForLog();
+            var containerClient = new BlobContainerClient(storageAccountConnectionString, containerName);
+
+            if (!await containerClient.ExistsAsync())
+            {
+                await containerClient.CreateAsync();
+            }
+
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(storageAccountConnectionString);
+            var container = new CloudBlobContainer(containerClient.Uri, storageAccount.Credentials);
+            return GetContainerSasUri(container);
+        }
+
+        internal static async Task UploadLogsAsync(string iotHubConnectionString, Uri blobContainerWriteUri, ILogger logger)
+        {
+            Preconditions.CheckNonWhiteSpace(iotHubConnectionString, nameof(iotHubConnectionString));
+            Preconditions.CheckNotNull(blobContainerWriteUri, nameof(blobContainerWriteUri));
+            Preconditions.CheckNotNull(logger, nameof(logger));
+
+            DateTime uploadLogStartAt = DateTime.UtcNow;
+            logger.LogInformation("Send upload logs request to edgeAgent.");
+
+            ModuleClient moduleClient = await ModuleClient.CreateFromEnvironmentAsync(TransportType.Amqp_Tcp_Only);
+            var uploadLogRequest = new MethodRequest(
+                "UploadLogs",
+                Encoding.UTF8.GetBytes($"{{ \"schemaVersion\": \"1.0\", \"sasUrl\": \"{blobContainerWriteUri.AbsoluteUri}\", \"items\": [{{ \"id\": \".*\", \"filter\": {{}} }}], \"encoding\": \"gzip\", \"contentType\": \"json\" }}"));
+            MethodResponse uploadLogResponse = await moduleClient.InvokeMethodAsync(Settings.Current.DeviceId, "$edgeAgent", uploadLogRequest);
+
+            (string status, string correlationId) = GetUploadLogResponseResult(uploadLogResponse.ResultAsJson);
+            logger.LogInformation($"Upload logs response: status={status}, correlationId={correlationId}");
+
+            int checkUploadStatusPeriod = 60 * 1000;    // 1 min
```

Comment:
Do we expect the blob storage upload to take a minute?

Reply:
it depends log file size and number of modules; since we have up to 20 modules and debug log level is used.  I can change it to 30s.